spec/javascripts/components/story/expanded_story/expanded_story_default/expanded_story_default_spec.js

```diff
@@ -1,77 +1,165 @@
-    createTask: vi.fn(),
-    deleteTask: vi.fn(),
-    toggleTask: vi.fn(),
-    deleteNote: vi.fn(),
-    createNote: vi.fn(),
-    addLabel: vi.fn(),
-    removeLabel: vi.fn(),
-    setLoadingStory: vi.fn(),
-    users: [],
-    project: { labels: [] },
-    enabled: true,
-    onClone: vi.fn(),
-    showHistory: vi.fn(),
-    disabled: false,
+  beforeEach(function () {
+    vi.spyOn(I18n, 't');
+  });
+
+  afterEach(function () {
+    I18n.t.mockRestore();
   });
 
-  it("renders all children components when isn't a new story", () => {
-    const story = {
-      ...storyFactory({ id: 42 }),
-      _editing: storyFactory({ id: 42 }),
+  vi.spyOn(window.md, 'makeHtml').mockReturnValue('<p>Test</p>');
+  vi.mock('react-clipboard.js', () => ({
+    default: vi
+      .fn()
+      .mockImplementation(({ children }) => <div>{children}</div>),
+  }));
+
+  const renderComponent = props => {
+    const defaultProps = {
+      story: {
+        ...storyFactory(),
+        _editing: storyFactory(),
+      },
+      onEdit: vi.fn(),
+      storyFailure: vi.fn(),
+      createTask: vi.fn(),
+      deleteTask: vi.fn(),
+      toggleTask: vi.fn(),
+      deleteNote: vi.fn(),
+      createNote: vi.fn(),
+      addLabel: vi.fn(),
+      removeLabel: vi.fn(),
+      setLoadingStory: vi.fn(),
+      users: [],
+      project: { pointValues: [], labels: [] },
+      enabled: true,
+      onClone: vi.fn(),
+      showHistory: vi.fn(),
+      disabled: false,
+    };
+    const mergedProps = { ...defaultProps, ...props };
+
+    return renderWithProviders(<ExpandedStoryDefault {...mergedProps} />);
+  };
+
+  it("renders all children components when story isn't new", () => {
+    const props = {
+      story: {
+        ...storyFactory({ id: 42 }),
+        _editing: storyFactory({ id: 42 }),
+      },
     };
 
-    const wrapper = shallow(
-      <ExpandedStoryDefault {...defaultProps()} story={story} />
+    const { container } = renderComponent(props);
+
+    // ExpandedStoryHistoryLocation
+    const expandedStoryHistoryLocation =
+      container.querySelector('#story-link-42');
+    expect(expandedStoryHistoryLocation).toBeInTheDocument();
+
+    // ExpandedStoryTitle
+    expect(I18n.t).toHaveBeenCalledWith('activerecord.attributes.story.title');
+
+    // ExpandedStoryEstimate
+
+    expect(I18n.t).toHaveBeenCalledWith(
+      'activerecord.attributes.story.estimate'
+    );
+
+    // ExpandedStoryType
+    expect(I18n.t).toHaveBeenCalledWith(
+      'activerecord.attributes.story.story_type'
+    );
+
+    // ExpandedStoryState
+    expect(I18n.t).toHaveBeenCalledWith('activerecord.attributes.story.state');
+
+    // ExpandedStoryRequestedBy
+    expect(I18n.t).toHaveBeenCalledWith(
+      'activerecord.attributes.story.requested_by'
+    );
+
+    // ExpandedStoryOwnedBy
+    expect(I18n.t).toHaveBeenCalledWith(
+      'activerecord.attributes.story.owned_by'
     );
 
-    expect(wrapper.find('ExpandedStoryHistoryLocation')).toExist();
```

Comment:
Haven't we applied the tests to check if the components are on the document? From the line 41 to 51

Reply:
Yeah, it's the closest approach. We can confirm the final render with `screen.debug()` after rendering the component.

The main issue with between the old tests and our migration is defintitely the ability to check for the JSX components in Enzyme, and how it translates into [testing-library's phylosophy](https://testing-library.com/docs/guiding-principles/).

> If it relates to rendering components, then it should deal with DOM nodes rather than component instances